### ISSUE TITLE
Driver set to 24bit but 16 bit audio is captured.

### ISSUE THREAD

**Issue by mzso** [rdp/screen-capture-recorder-to-video-windows-free#52]
Hello!

Everything else is as it should be. The sampling rate and the channel numbers are exactly what I set, only this is different.
Is it a bug? Any way to work around it?

**Comment by rdp** [comment]
difference between 24 bit and 16 bit is practically nothing AFAIK...plus 24
bit is "less compatible" (I don't think many encoders can handle it, they
require 16 bit) is 16 bit ok?

**Comment by mzso** [comment]
@rdp commented on [2015. jan. 1. 3:02 CET](https://github.com/rdp/screen-capture-recorder-to-video-windows-free/issues/52#issuecomment-68478431 "2015-01-01T02:02:27Z - Replied by Github Comment Enhancer"):
> <div class="email-fragment">difference between 24 bit and 16 bit is practically nothing AFAIK...plus 24
> bit is "less compatible" (I don't think many encoders can handle it, they
> require 16 bit) is 16 bit ok?</div>

Well, the way I see it it's the capture software or encoders job to manipulate the raw output into it's final form. Could be 128k mp3 or 24 bit flac, depending on what's desired.

I didn't have any problems with 24 bit audio. All encoders handled it without issue. (Flac, tak, opus, vorbis, mp3)

I don't know what you mean by "ok".

**Comment by mca64** [comment]
rdp could you add original audio depth? The same as speakers, not always 16 bit

**Comment by rdp** [comment]
The problem I ran into when testing is that if I let through something
besides 16 bit, then certain encoders like flash media encoder would
*reject* the stream...hmm...when I researched it it seemed that everybody
agreed there was not much difference in quality between 24 and 16 bit [the
equivalent of "white noise" as it were...hmm...

**Comment by mzso** [comment]
As far as I know there's some chance that there's audible difference. 16 bit's 96dB dynamic range doesn't surpass human hearing's. Also there might be quantization noise without dithering.
Maybe these aren't much of issues. But if people want to extensively process the audio, it's usually claimed that more bits are better.

Also not recording the audio as-it-is is just yet another conversion, which can only take away from the quality.

(BTW does SCR have any sort of configurability? )

**Comment by rdp** [comment]
There might be some way to "allow" for 24 bit while defaulting to 16 bit [I
think it's still probably a good idea to default to 16 bit, otherwise it
won't work with flash media encoder at all...as far as I
remember]...anyway, I'll put it on the TODO list (funding welcome, patch
welcome).
Thanks for the note, I'll leave this issue open.
Thanks!

**Comment by rdp** [comment]
SCR you can specify the bit depth you want, if I remember correctly
[default is either 32 bit or 24 I can't remember]

**Comment by mca64** [comment]
"difference between 24 bit and 16 bit is practically nothing AFAIK..."  you should say that, espescially that you are author of some audio toolf. Its kinda lame, like saying why you need flac where mp3 320 is practically the same.  but who cares ;)

**Comment by mca64** [comment]
btw for capturing 24 bits does matter, more than high freq

**Comment by mca64** [comment]
http://www.soundonsound.com/sos/jun08/articles/qa0608_2.htm

**Comment by rdp** [comment]
Yeah I'm not saying it's a bad idea, I am saying it's somewhat of a lesser
priority, though, patches welcome!

**Comment by rdp** [comment]
If you go back to the commit before this one, it will have 24 bit audio,
you could build one, GL!
https://github.com/rdp/virtual-audio-capture-grabber-device/commit/8ff9f6542a52a85327f69ab93a23d25e5a47e306

**Comment by mzso** [comment]
What I don't understand is why "flash media live encoder" is a reason to
anything.
It's one piece of software from Adobe. Which if I go by what I know of
their other software is probably not of qood quality.

2015-01-04 2:00 GMT+01:00 Roger Pack <notifications@github.com>:

> If you go back to the commit before this one, it will have 24 bit audio,
> you could build one, GL!
>
> https://github.com/rdp/virtual-audio-capture-grabber-device/commit/8ff9f6542a52a85327f69ab93a23d25e5a47e306
>

**Comment by mzso** [comment]
@rdp commented on [2015. jan. 2. 17:36 CET](https://github.com/rdp/screen-capture-recorder-to-video-windows-free/issues/52#issuecomment-68540037 "2015-01-02T16:36:24Z - Replied by Github Comment Enhancer"):
> <div class="email-fragment">SCR you can specify the bit depth you want, if I remember correctly
> [default is either 32 bit or 24 I can't remember]</div>
> </div>
> </div>

I actually was thinking of audio. I thought it was considered the same piece of software.
What I really liked to know if how to change whatever settings available. I didn't find anything to do so.

**Comment by mzso** [comment]
So, this is still the case. No chance for at least an option to record the stream as-it-is from the system?

**Comment by rdp** [comment]
funding welcome :)
